From 2581b9a2377ae411392ebdf56fbccb9eaed3b3d6 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Thu, 29 Apr 2021 11:20:25 +0200
Subject: [PATCH] reintroduce cqhci_suspend for kABI
Patch-mainline: Never, kABI fixup
References: git-fixes

Basically undo the patch introducing deactivate()
and duplicating the symbol.

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/mmc/host/cqhci.c | 7 +++++++
 drivers/mmc/host/cqhci.h | 5 +----
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/cqhci.c b/drivers/mmc/host/cqhci.c
index 2d65b32d2..a879a2fb1 100644
--- a/drivers/mmc/host/cqhci.c
+++ b/drivers/mmc/host/cqhci.c
@@ -1148,6 +1148,13 @@ int cqhci_init(struct cqhci_host *cq_host, struct mmc_host *mmc,
 }
 EXPORT_SYMBOL(cqhci_init);
 
+int cqhci_suspend(struct mmc_host *mmc)
+{
+	return cqhci_deactivate(mmc);
+}
+
+EXPORT_SYMBOL(cqhci_suspend);
+
 MODULE_AUTHOR("Venkat Gopalakrishnan <venkatg@codeaurora.org>");
 MODULE_DESCRIPTION("Command Queue Host Controller Interface driver");
 MODULE_LICENSE("GPL v2");
diff --git a/drivers/mmc/host/cqhci.h b/drivers/mmc/host/cqhci.h
index 437700179..e062b42ad 100644
--- a/drivers/mmc/host/cqhci.h
+++ b/drivers/mmc/host/cqhci.h
@@ -231,10 +231,7 @@ irqreturn_t cqhci_irq(struct mmc_host *mmc, u32 intmask, int cmd_error,
 int cqhci_init(struct cqhci_host *cq_host, struct mmc_host *mmc, bool dma64);
 struct cqhci_host *cqhci_pltfm_init(struct platform_device *pdev);
 int cqhci_deactivate(struct mmc_host *mmc);
-static inline int cqhci_suspend(struct mmc_host *mmc)
-{
-	return cqhci_deactivate(mmc);
-}
+int cqhci_suspend(struct mmc_host *mmc);
 int cqhci_resume(struct mmc_host *mmc);
 
 #endif
-- 
2.26.2

